WordPress/gutenberg
# Normalize converting nav menu items to blocks between Editor and Block

## Description
The aim of this PR is to normalize the _implementations_ for converting menu items to block attributes between Nav block and Nav editor and then ensure the code is correctly duplicated across both implementations.

The wider goal is that once we have the implementation normalized we can more easily [extract it to a shared utility in the proposed `@wordpress/menus` package](https://github.com/WordPress/gutenberg/issues/31580). Note I want to tackle the normalization in a separate PR in order to make the process of creating and merging the new _package_ easier. Trying to everything in a single large PR would make the introduction of the package harder than it already is.

Note the work in this PR has benefits outside of the introduction of the package namely:

* consistency
* test coverage
* ease of maintenance.

...etc.

As a side consequence of this PR the Nav Block will now respect menu _ordering_ if creating a Nav block from an existing WP Menu.

This PR:

- [x] updates the Nav block mapping utilities to cater for sorting converted menu items by `menu_order`.
- [x] adds the ability to build a menuItem -> block.clientId as part of the mapping function return signature.
- [x] duplicates the enhanced Nav Block mapping implementation to the Editor.
- [x] removes custom mapping functionality in the Editor.
- [x] adds tests

Closes https://github.com/WordPress/gutenberg/issues/31599

## How has this been tested?




We have new unit tests to provide some level of coverage for the block mapping functions - one for each duplicate in the Nav Editor and Block:

* packages/block-library/src/navigation/test/menu-items-to-blocks.js
* packages/edit-navigation/src/store/test/menu-items-to-blocks.js

To test this manually you should confirm that the editor and block behave as they do on `trunk`. Specifically try to focus on 

* nested menus.
* menu ordering.
* creation of additional menu items and saving (Navigation Editor only).


To do this I first recommend creating a big old nested menu using the existing menus screen (`wp-admin/nav-menus.php`). Go wild and nest items as far down as you want! Whilst you're doing this make a note of menu ordering.

* Open the Navigation editor and cross check that all the items from your menu came through and that they are **ordered** and **nested** correctly.
* In the editor also check that you can add a few _new_ menu items at different nesting levels and have that saved back faithfully to the existing Menus screen at `wp-admin/nav-menus.php`.
* Repeat for the Block - you'll need to "Create from existing menu" on the placeholder and select the nested menu you created. Again cross-check that all the items from your menu came through and that they are **ordered** and **nested** correctly.

## Screenshots 

## Types of changes





## Checklist:
